compiler/noirc_frontend/src/elaborator/expressions.rs

```diff
@@ -75,7 +75,10 @@ impl<'context> Elaborator<'context> {
                 self.push_err(ResolverError::UnquoteUsedOutsideQuote { span: expr.span });
                 (HirExpression::Error, Type::Error)
             }
-            ExpressionKind::AsTraitPath(_) => todo!("Implement AsTraitPath"),
+            ExpressionKind::AsTraitPath(_) => {
+                self.push_err(ResolverError::UnquoteUsedOutsideQuote { span: expr.span });
+                (HirExpression::Error, Type::Error)
```

Comment:
Should a different error be issued here?

Reply:
Yep, this should be reverted, it's remaining from debugging code

Reply:
Sounds good. I think this panic makes LSP crash, so even if you don't want to write an `AsTraitPath` it might happen if it's an incomplete expression. It would be nice to at least change this panic to a user-error (in a separate PR, of course).

Reply:
That's how this was originally triggering while I was debugging as well.

When I was working on the type alias example in slack:

```noir
type Double<let N: u32> = N * 2;

fn main() {
    let _x: [Field; Double<2>] = [1, 2, 3, 4];
}
```
We were getting a syntax error on Double<2> and automatically correcting it to an AsTraitPath